bin/ch/ch.cpp

```diff
@@ -79,77 +79,50 @@ void __stdcall PrintUsage()
 {
     HRESULT hr = S_OK;
-    *byteCodeBuffer = nullptr;
-    *byteCodeBufferSize = 0;
-    BYTE *bcBuffer = nullptr;
 
-    unsigned int bcBufferSize = 0;
-    unsigned int newBcBufferSize = 0;
     JsValueRef scriptSource;
     IfJsErrorFailLog(ChakraRTInterface::JsCreateExternalArrayBuffer((void*)fileContents,
         (unsigned int)strlen(fileContents), nullptr, nullptr, &scriptSource));
-    IfJsErrorFailLog(ChakraRTInterface::JsSerialize(scriptSource, bcBuffer,
-        &bcBufferSize, JsParseScriptAttributeNone));
-    // Call above will return the size of the buffer only, once succeed
-    // we need to allocate memory of that much and call it again.
-    if (bcBufferSize == 0)
-    {
-        AssertMsg(false, "bufferSize should not be zero");
-        IfFailGo(E_FAIL);
-    }
-    bcBuffer = new BYTE[bcBufferSize];
-    newBcBufferSize = bcBufferSize;
-    IfJsErrorFailLog(ChakraRTInterface::JsSerialize(scriptSource, bcBuffer,
-        &newBcBufferSize, JsParseScriptAttributeNone));
-    Assert(bcBufferSize == newBcBufferSize);
+    IfJsErrorFailLog(ChakraRTInterface::JsSerialize(scriptSource, byteCodeBuffer,
+        JsParseScriptAttributeNone));
 
 Error:
-    if (hr != S_OK)
-    {
-        // In the failure release the buffer
-        if (bcBuffer != nullptr)
-        {
-            delete[] bcBuffer;
-        }
-    }
-    else
-    {
-        *byteCodeBuffer = bcBuffer;
-        *byteCodeBufferSize = bcBufferSize;
-    }
-
     return hr;
 }
 
 HRESULT CreateLibraryByteCodeHeader(LPCSTR contentsRaw, DWORD lengthBytes, LPCWSTR bcFullPath, LPCSTR libraryNameNarrow)
 {
     HANDLE bcFileHandle = nullptr;
+    JsValueRef bufferVal;
     BYTE *bcBuffer = nullptr;
-    DWORD bcBufferSize = 0;
-    HRESULT hr = GetSerializedBuffer(contentsRaw, &bcBuffer, &bcBufferSize);
-
-    if (FAILED(hr)) return hr;
-
-    bcFileHandle = CreateFile(bcFullPath, GENERIC_WRITE, FILE_SHARE_DELETE, nullptr, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr);
-    if (bcFileHandle == INVALID_HANDLE_VALUE)
-    {
-        return E_FAIL;
-    }
-
+    unsigned int bcBufferSize = 0;
     DWORD written;
-
     // For validating the header file against the library file
     auto outputStr =
         "//-------------------------------------------------------------------------------------------------------\r\n"
         "// Copyright (C) Microsoft. All rights reserved.\r\n"
         "// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.\r\n"
         "//-------------------------------------------------------------------------------------------------------\r\n"
         "#if 0\r\n";
+
+
+    HRESULT hr = GetSerializedBuffer(contentsRaw, &bufferVal);
+
+    if (FAILED(hr)) return hr;
+
+    IfFalseGo(ChakraRTInterface::JsGetArrayBufferStorage(bufferVal, &bcBuffer,
+        &bcBufferSize) != JsNoError);
+
+    bcFileHandle = CreateFile(bcFullPath, GENERIC_WRITE, FILE_SHARE_DELETE,
```

Comment:
Sounds like a nice improvement in general. Could you please create a Github issue?